packages/common/src/search/types/IHubSearchResult.ts

```diff
@@ -0,0 +1,110 @@
+import { IHubGeography } from "../..";
+
+/**
+ * Standardized light-weight search result structure, applicable to all
+ * types of search results - users, groups, content, events etc
+ */
+export interface IHubSearchResult {
+  /**
+   * Access level of the backing entity
+   */
+  access: "public" | "shared" | "org" | "private";
+  /**
+   * Identifier - could be a guid or a username
+   */
+  id: string;
+  /**
+   * Hub Family: "dataset" | "map" | "document" | "content"
+   * TODO: Can we create a type for this?
+   */
+  family: string;
+  /**
+   * Actual type
+   * For Item backed results, this will be item.type
+   * Otherwise it will be "Group", "User", "Event" etc
+   */
+  type: string;
+  /**
+   * Name or title
+   */
+  name: string;
+  /**
+   * Owner; Applies to Items and Groups
+   */
+  owner: string;
+  /**
+   * Sanitized summary derived from item.snippet, item.description, group.description
+   */
+  summary: string;
+  /**
+   * Date when the entity was created
+   */
+  createdDate: Date;
+  /**
+   * Source of the creation date as a property path
+   * e.g. `item.created`
+   */
+  createdDateSource: string;
+  /**
+   * Date when the entity was last updated
+   * Depending on the entity, this could be derived
+   * in many different ways
+   */
+  updatedDate?: Date;
+  /**
+   * Source of the updated date as a property path
+   * e.g. `featureService.lastEdit`
+   */
+  updatedDateSource?: string;
+  /**
+   * Fully qualified thumbnail url for items, groups, users
+   * Will not have the token appended. Consuming app needs to
+   */
+  thumbnailUrl: string;
+  /**
+   * Hash of pre-computed urls
+   */
+  urls: {
+    /**
+     * Canononical ArcGIS Online Url for this entity (if applicable)
+     */
+    agoUrl?: string;
+    /**
+     * Relative url, which can be appended to a Site's root Url
+     * e.g. /datasets/3ef8c7a, or /events/june-meeting
+     * This is much easier than passing a Site into the search
+     * system so the full url can be computed
+     */
+    siteRelativeUrl: string;
+  };
+  /**
+   * Used in the Gallery components, where a callback can be
+   * provided to compute a custom url for each result.
+   * Useful when using those components in a custom app and
+   * the routing is different from urls.siteRelativeUrl
+   */
+  customUrl?: string;
+  /**
+   * Geometry connected to this entity
+   * For items, it will default to the extent,
+   * but may be derived from a boundary resource
+   * or the extent of a layer
+   */
+  geometry?: IHubGeography;
```

Comment:
we use `boundary` for this in `IHubContent`. I'm not sure how I feel about that name to begin with, and I can see why that term is even less desirable for users, etc. However, I don't like `geometry` b/c that implies it's a geometry that I can pass to the JSAPI. In the absence of something compelling that works for all entities, I'd say stick w/ `boundary`.

Reply:
I'm interested in @ajturner's input on this, particularly re: thinking about User Geographies (?? I forget the exact terminology though) - I think we want this to be applicable to anything, and be a point/line/poly or a set of any of the above.

Reply:
My original design was [`geography` with a type that included the geometry as well as location metadata](https://github.com/esridc/hub-components-experiments/blob/master/src/utils/hub-types.ts#L175) such as name, coverage, source, etc. for example the geometery likely could be from a feature service of admin boundaries - copied here but we want to attribute the source and possibly refresh from source when it changes

I do plan for Users and Groups to have multiple geography - and that could be true for any entity (multiple areas for a project, event venues, etc.)